app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,25 @@
+class CommentsController < ApplicationController
+  def new
+    @comment = Comment.new
+    respond_to do |format|
+      format.html { render :new, locals: { comment: @comment } }
+    end
+  end
+
+  def create
+    @comment = Comment.new(params.require(:comment).permit(:text))
+    @comment.post_id = params[:post_id]
+    @comment.author_id = current_user.id
+    respond_to do |format|
+      format.html do
+        if @comment.save
+          flash[:success] = 'Comment created succesfully'
+          redirect_to user_post_path(current_user, @comment.post_id)
+        else
+          flash[:error] = 'Error: Not able to create post'
+          render :new, locals: { comment: @comment }
```

Comment:
## OPTIONAL
* I strongly suggest you add a `comment_params ` method, This is a better security practice to help prevent accidentally allowing users to update sensitive model attributes. [Read more here about strong params](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides)

Reply:
I'm applying your suggestion tomorrow😄